src/Engine/Image.php

```diff
@@ -0,0 +1,76 @@
+<?php
+declare(strict_types = 1);
+
+namespace App\Engine;
+
+use Imagine\Filter\Basic\Crop;
+use Imagine\Image\Box;
+use Imagine\Image\Point;
+
+class Image
+{
+
+    /** @var string Original URL of image. */
+    private $imageUrl;
+
+    /** @var int[] Array of floats with keys: `x`, `y`, `width`, and `height`. */
+    private $crop;
+
+    /** @var string Image data. */
+    private $data;
+
+    /** @var int Image size in bytes. */
+    private $size;
+
+    /**
+     * @param int[] $crop
+     */
+    public function __construct(string $imageUrl, array $crop = [])
+    {
+        $this->imageUrl = $imageUrl;
+        $this->crop = $crop;
+    }
+
+    public function geturl(): string
+    {
+        return $this->imageUrl;
+    }
+
+    public function needsCropping(): bool
+    {
+        return isset($this->crop['width']) && $this->crop['width'] > 0
+            && isset($this->crop['height']) && $this->crop['height'] > 0;
+    }
+
+    public function getCrop(): Crop
+    {
+        return new Crop(
+            new Point($this->crop['x'], $this->crop['y']),
+            new Box($this->crop['width'], $this->crop['height'])
+        );
+    }
+
+    public function getData(): string
+    {
+        return $this->data;
+    }
+
+    public function setData(string $data): void
+    {
+        $this->data = $data;
+    }
+
+    /**
+     * Get the image data size in bytes.
+     * @return int
+     */
+    public function getSize(): int
+    {
+        return $this->size ?? strlen($this->data);
```

Comment:
This is also implicitly assuming that $this->data is set if $this->size is not. I think we could either enforce calling getSize() (by throwing if $this->size is null), or changing this to

```
if ( $this->size !== null ) {
    return $this->size;
}
if ( $this->data !== null ) {
    return strlen( $this->data );
}

throw ...
```

Reply:
Yeah, especially if we agree that in the URL case, size means "size of the data pointed by the URL".